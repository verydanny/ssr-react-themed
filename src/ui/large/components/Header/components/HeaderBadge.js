import React, { PureComponent } from 'react'
import themed from 'react-themed-too'
import PropTypes from 'prop-types'

@themed(/Header/)
class HeaderBadge extends PureComponent {

  static propTypes = {
    theme: PropTypes.object,
    numFavorites: PropTypes.number.isRequired,
  }

  static defaultProps = {
    theme: {},
  }

  render() {
    const { theme, numFavorites } = this.props

    return (
      <span className={theme.Header_Badge} data-tid="favorites-badge">
        {numFavorites}
      </span>
    )
  }
}

export default HeaderBadge
