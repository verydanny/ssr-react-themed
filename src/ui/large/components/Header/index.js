export Header from './Header'
export HomeHeader from './HomeHeader'
