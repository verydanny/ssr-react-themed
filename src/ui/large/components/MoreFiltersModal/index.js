export { default as MoreFiltersModal } from './container'
export { MORE_FILTERS_MODAL_ID } from './const'
