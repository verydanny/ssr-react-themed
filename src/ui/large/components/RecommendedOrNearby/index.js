export { default as RecommendedOrNearby } from './container'
