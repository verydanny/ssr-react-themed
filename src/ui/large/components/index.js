export MapView from './MapView'
export ListView from './ListView'
export Layout from './Layout'
export HybridMap from './HybridMap'
