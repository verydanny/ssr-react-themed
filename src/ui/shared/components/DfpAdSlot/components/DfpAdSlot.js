import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import { AdSlot, DFPManager } from 'react-dfp'
import themed from 'react-themed-too'
import cn from 'classnames'
import DfpAds from 'config/dfpAds'
import { stateAbbreviation } from 'lib/geo/states'

@themed(/^Dfp/)

export default class DfpAdSlot extends PureComponent {
  static propTypes = {
    adKey: PropTypes.string.isRequired,
    customTargeting: PropTypes.object,
    doNotRefresh: PropTypes.bool,
    onSlotRender: PropTypes.func,
    className: PropTypes.string,
    city: PropTypes.object,
    state: PropTypes.object,
    zip: PropTypes.string,
    lat: PropTypes.number,
    lng: PropTypes.number,
    targetCode: PropTypes.string,
    tid: PropTypes.string,
    theme: PropTypes.object,
    bypass: PropTypes.bool,
  }

  static defaultProps = {
    customTargeting: null,
    doNotRefresh: false,
    bypass: false,
    city: {},
    state: {},
    theme: {},
  }

  static contextTypes = {
    dfp: PropTypes.object,
  }

  constructor(props) {
    super(props)
    this.state = { ready: false, mounted: false }
  }

  componentDidMount() {
    if (!this.props.bypass) DFPManager.load()

    this.updateMountState()
  }

  get mappings() {
    const { city, state, zip, lat, lng, targetCode } = this.props

    return {
      city: city.name,
      state: stateAbbreviation(state.name),
      zip,
      latitude: lat,
      longitude: lng,
      targetCode,
    }
  }

  get slotRenderCallback() {
    const { onSlotRender, adKey } = this.props
    const { setDisplayReady } = this.context.dfp || {}

    return (gpt = { event: {} }) => {
      // googletag fires slot render event even if ad is empty,
      // so explicit check is needed here
      if (!gpt.event.isEmpty) {
        this.setState({ ready: true })
        if (onSlotRender) onSlotRender(adKey)
        if (setDisplayReady) setDisplayReady()
      }
    }
  }

  get ads() {
    return DfpAds.ads || {}
  }

  get isDisabled() {
    return DfpAds.isDisabled()
  }

  get baseTargeting() {
    const baseObj = {}
    const { adKey } = this.props
    const { targets = [] } = this.ads[adKey]

    if (targets.length) {
      targets.forEach(key => {
        baseObj[key] = this.mappings[key]
      })
    }

    return baseObj
  }

  get targetingArguments() {
    const tArgs = this.filterObject({
      ...this.baseTargeting,
      ...this.props.customTargeting,
    })

    return Object.keys(tArgs).length ? tArgs : null
  }

  updateMountState() {
    this.setState({ mounted: true })
  }

  filterObject(obj) {
    const filtered = Object.assign({}, obj)
    Object.keys(filtered).forEach(key => {
      const val = filtered[key]

      if (val === null || val === undefined) {
        delete filtered[key]
      }
    })
    return filtered
  }

  render() {
    const { adKey, doNotRefresh, className, theme, tid, bypass } = this.props

    if (!this.ads[adKey]) return null

    const { sizes, network, adUnit, mute } = this.ads[adKey]

    if (!this.state.mounted || mute || this.isDisabled || bypass) return null
    return (
      <div
        className={cn({
          [theme.Dfp]: true,
          [theme['Dfp-centered']]: true,
          [theme['Dfp-ready']]: this.state.ready,
          [className]: className,
        })}
        data-tid={tid}
        id={adKey}
      >
        <AdSlot
          sizes={sizes}
          dfpNetworkId={network}
          adUnit={adUnit}
          shouldRefresh={() => !doNotRefresh}
          targetingArguments={this.targetingArguments}
          onSlotRender={this.slotRenderCallback}
        />
      </div>
    )
  }
}
