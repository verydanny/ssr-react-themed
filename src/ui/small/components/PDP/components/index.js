export PDPAddress from './PDPAddress'
export PDPAmenities from './PDPAmenities'
export PDPFooter from './PDPFooter'
export PDPGallery from './PDPGallery'
export PDPHeader from './PDPHeader'
export PDPKeyInfo from './PDPKeyInfo'
export PDPLayout from './PDPLayout'
export PDPLeasingTerms from './PDPLeasingTerms'
export PDPMap from './PDPMap'
export PDPRatings from './PDPRatings'
export PDPPetPolicy from './PDPPetPolicy'
export PDPReview from './PDPReview'
export PDPReviewGallery from './PDPReviewGallery'
export PDPFloorplans from './PDPFloorplans'
