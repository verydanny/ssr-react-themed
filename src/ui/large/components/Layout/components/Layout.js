import React from 'react'
import PropTypes from 'prop-types'
import ModalStack from 'ui/shared/components/ModalStack'
import { modalDefinitions } from 'ui/large/config/modalConfig'
import { Header as DefaultHeader } from 'ui/large/components/Header'
import { themed } from 'react-themed-too'

const Layout = props => (
  <div className={props.theme.Layout_Container}>
    <ModalStack modalDefinitions={modalDefinitions} />
    {props.HeaderComponent && <props.HeaderComponent />}
    <section className={props.theme.Layout_Content}>
      { props.children }
    </section>
  </div>
)

Layout.propTypes = {
  children: PropTypes.node,
  HeaderComponent: PropTypes.any,
  theme: PropTypes.object,
}

Layout.defaultProps = {
  theme: {},
  HeaderComponent: DefaultHeader,
}

export default themed(/^Layout/)(Layout)
