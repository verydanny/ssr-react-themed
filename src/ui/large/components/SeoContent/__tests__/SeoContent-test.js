import React from 'react'
import { shallow } from 'enzyme'
import { DfpAdSlot } from 'ui/shared/components/DfpAdSlot'
import WrappedSeoContent from '../SeoContent'

const SeoContent = WrappedSeoContent.WrappedComponent

describe('SeoContent', () => {
  const prop = { links: [], maxLinksShowing: 5 }
  const props = {
    apartmentLinks: [],
    propertyTypeLinks: prop,
    cityLinks: prop,
    neighborhoods: prop,
    localInfoBody: 'localInfoBody',
    propertyType: 'propertyType',
    colleges: [],
    militaryBases: [],
    shouldDisplayLocalInfo: true,
    theme: {},
  }

  const setupShallow = overrideProps => {
    const setupProps = { ...props, ...overrideProps }
    return shallow(<SeoContent {...setupProps} />)
  }

  describe('apartmentLinks or propertyTypeLinks', () => {
    let wrapper

    beforeEach(() => {
      wrapper = setupShallow()
    })

    it('does not render if there are none', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('localInfoBody', () => {
    describe('when shouldDisplayLocalInfo is true', () => {
      it('renders', () => {
        const wrapper = setupShallow({ shouldDisplayLocalInfo: true })
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('when shouldDisplayLocalInfo is false', () => {
      it('does not render', () => {
        const wrapper = setupShallow({ shouldDisplayLocalInfo: false })
        expect(wrapper).toMatchSnapshot()
      })
    })
  })

  describe('nearby colleges', () => {
    it('displays nearby colleges if shouldDisplayNearbyColleges is true', () => {
      const wrapper = setupShallow({ shouldDisplayNearbyColleges: true })
      expect(wrapper).toMatchSnapshot()
    })

    it('does not display nearby colleges if shouldDisplayNearbyColleges is false', () => {
      const wrapper = setupShallow({ shouldDisplayNearbyColleges: false })
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('advertisement', () => {
    it('appears on the page', () => {
      const wrapper = setupShallow()
      expect(wrapper.find(DfpAdSlot)).toHaveLength(2)
    })
  })
})
