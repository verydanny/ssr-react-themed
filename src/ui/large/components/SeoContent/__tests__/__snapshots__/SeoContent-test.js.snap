// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeoContent apartmentLinks or propertyTypeLinks does not render if there are none 1`] = `
<div>
  <div
    data-tid="seo-content"
  >
    <div />
    <Connect(Themed(DfpAdSlot))
      adKey="gpt_list_srp_desktop_bottom"
      tid="srp-bottom-ad"
    />
    <Themed(LocalInfo)
      body="localInfoBody"
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType in Nearby Cities"
      linkTagItem="by_nearby_cities"
      linkTagSection="local_info"
      maxLinks={5}
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType in Nearby Neighborhoods"
      linkTagItem="by_nearby_neighborhoods"
      linkTagSection="local_info"
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType Near Local Military Bases"
      linkTagItem="by_nearby_military"
      linkTagSection="local_info"
    />
  </div>
  <Connect(Themed(DfpAdSlot))
    adKey="gpt_list_srp_desktop_bottom_ad2"
    tid="srp-bottom-ad2"
  />
</div>
`;

exports[`SeoContent localInfoBody when shouldDisplayLocalInfo is false does not render 1`] = `
<div>
  <div
    data-tid="seo-content"
  >
    <div />
    <Connect(Themed(DfpAdSlot))
      adKey="gpt_list_srp_desktop_bottom"
      tid="srp-bottom-ad"
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType in Nearby Cities"
      linkTagItem="by_nearby_cities"
      linkTagSection="local_info"
      maxLinks={5}
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType in Nearby Neighborhoods"
      linkTagItem="by_nearby_neighborhoods"
      linkTagSection="local_info"
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType Near Local Military Bases"
      linkTagItem="by_nearby_military"
      linkTagSection="local_info"
    />
  </div>
  <Connect(Themed(DfpAdSlot))
    adKey="gpt_list_srp_desktop_bottom_ad2"
    tid="srp-bottom-ad2"
  />
</div>
`;

exports[`SeoContent localInfoBody when shouldDisplayLocalInfo is true renders 1`] = `
<div>
  <div
    data-tid="seo-content"
  >
    <div />
    <Connect(Themed(DfpAdSlot))
      adKey="gpt_list_srp_desktop_bottom"
      tid="srp-bottom-ad"
    />
    <Themed(LocalInfo)
      body="localInfoBody"
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType in Nearby Cities"
      linkTagItem="by_nearby_cities"
      linkTagSection="local_info"
      maxLinks={5}
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType in Nearby Neighborhoods"
      linkTagItem="by_nearby_neighborhoods"
      linkTagSection="local_info"
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType Near Local Military Bases"
      linkTagItem="by_nearby_military"
      linkTagSection="local_info"
    />
  </div>
  <Connect(Themed(DfpAdSlot))
    adKey="gpt_list_srp_desktop_bottom_ad2"
    tid="srp-bottom-ad2"
  />
</div>
`;

exports[`SeoContent nearby colleges displays nearby colleges if shouldDisplayNearbyColleges is true 1`] = `
<div>
  <div
    data-tid="seo-content"
  >
    <div />
    <Connect(Themed(DfpAdSlot))
      adKey="gpt_list_srp_desktop_bottom"
      tid="srp-bottom-ad"
    />
    <Themed(LocalInfo)
      body="localInfoBody"
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType in Nearby Cities"
      linkTagItem="by_nearby_cities"
      linkTagSection="local_info"
      maxLinks={5}
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType in Nearby Neighborhoods"
      linkTagItem="by_nearby_neighborhoods"
      linkTagSection="local_info"
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType Near Local Colleges"
      linkTagItem="by_nearby_colleges"
      linkTagSection="local_info"
      maxLinks={10}
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType Near Local Military Bases"
      linkTagItem="by_nearby_military"
      linkTagSection="local_info"
    />
  </div>
  <Connect(Themed(DfpAdSlot))
    adKey="gpt_list_srp_desktop_bottom_ad2"
    tid="srp-bottom-ad2"
  />
</div>
`;

exports[`SeoContent nearby colleges does not display nearby colleges if shouldDisplayNearbyColleges is false 1`] = `
<div>
  <div
    data-tid="seo-content"
  >
    <div />
    <Connect(Themed(DfpAdSlot))
      adKey="gpt_list_srp_desktop_bottom"
      tid="srp-bottom-ad"
    />
    <Themed(LocalInfo)
      body="localInfoBody"
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType in Nearby Cities"
      linkTagItem="by_nearby_cities"
      linkTagSection="local_info"
      maxLinks={5}
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType in Nearby Neighborhoods"
      linkTagItem="by_nearby_neighborhoods"
      linkTagSection="local_info"
    />
    <Themed(Nearby)
      data={Array []}
      header="Find propertyType Near Local Military Bases"
      linkTagItem="by_nearby_military"
      linkTagSection="local_info"
    />
  </div>
  <Connect(Themed(DfpAdSlot))
    adKey="gpt_list_srp_desktop_bottom_ad2"
    tid="srp-bottom-ad2"
  />
</div>
`;
